site/app/models/gradeable/Gradeable.php

```diff
@@ -530,12 +530,50 @@ public function setPeerGradersList($input) {
         else {
             $this->core->getQueries()->clearPeerGradingAssignment($this->getId());
             foreach ($input as $row_num => $vals) {
-                $this->core->getQueries()->insertPeerGradingAssignment($vals["grader"], $vals["student"], $this->getId());
+                $this->core->getQueries()->insertPeerGradingAssignment($vals["grader"], $vals["student"], $vals["feedback"]);
                 $this->modified = true;
                 $this->peer_grading_pairs = $this->core->getQueries()->getPeerGradingAssignment($this->getId());
             }
         }
     }
+    
+    public function setPeerFeedback($grader_id, $student_id, $feedback) {
+        $bad_input = [];
+        if ($this->core->getQueries()->getUserById($grader_id) === null) {
+            array_push($bad_input, ($grader_id));
+        }
+        if ($this->core->getQueries()->getUserById($student_id) === null) {
+            array_push($bad_input, ($student_id));
+        }
+        if (!empty($bad_input)) {
+            $msg = "The given user id is not valid: ";
+            array_walk($bad_input, function ($val) use (&$msg) {
+                $msg .= " {$val}";
+            });
+            $this->core->addErrorMessage($msg);
+        }
+        else {
+            $this->core->getQueries()->insertPeerGradingFeedback($grader_id, $student_id, $this->getId(), $feedback);
+        }
+    }
+    
+    public function getPeerFeedback($grader_id, $anon_id) {
+        $user_id = $this->core->getQueries()->getUserFromAnon($anon_id)[$anon_id];
+        $feedback = $this->core->getQueries()->getPeerFeedbackInstance($this->getId(), $grader_id, $user_id);
+        if ($feedback == 'thanks') {
+            return 'Thank you!';
+        }
+        elseif ($feedback == 'helpful') {
+            return 'This feedback was helpful to me!';
+        }
+        elseif ($feedback == 'detailed') {
+            return 'This feedback was detailed, specific, and/or technical';
+        }
+        elseif ($feedback == 'inappropriate') {
+            return 'This feedback was inaccurate and/or inappropriate';
+        }
+        return 'No response';
+    }
```

Comment:
I don't think this function is ever called.

Reply:
It is called in RubricPanel.twig in the following line 

Peer Feedback: {{gradeable.getPeerFeedback(grader_id, anon_id)}}. This is where we translate the database entry to twig output, since we are no longer storing both in the table